diff --git a/views/layout/default.blade.php b/views/layout/default.blade.php
index fceaaa0..bfc76e6 100644
--- a/views/layout/default.blade.php
+++ b/views/layout/default.blade.php
@@ -87,7 +87,7 @@
 		Grocy.Components = { };
 		Grocy.Mode = '{{ GROCY_MODE }}';
-		Grocy.BaseUrl = '{{ $U('/') }}';
-		Grocy.CurrentUrlRelative = "/" + window.location.href.split('?')[0].replace(Grocy.BaseUrl, "");
+		Grocy.BaseUrl = window.location.origin + '{{ $U('/') }}';
+		Grocy.CurrentUrlRelative = "/" + window.location.pathname.replace(Grocy.BaseUrl, "");
 		Grocy.ActiveNav = '@yield('activeNav', '')';
 		Grocy.Culture = '{{ GROCY_LOCALE }}';
 		Grocy.Currency = '{{ GROCY_CURRENCY }}';
